mongodb/intro/date-aggregation/aggregation-with-date-expression-operators-iii.md

```diff
@@ -0,0 +1,169 @@
+---
+author: Stefan-Stojanovic
+
+levels:
+  - beginner
+  - basic
+  
+aspects:
+  - introduction
+  - new
+
+type: normal
+
+category: how to
+
+---
+
+# Aggregation With $hour, $minutes, $seconds and $milliseconds
+
+---
+## Content
+
+For the document used in aggregation, make sure to check the Previous Document[1] footnote.
+
+### $hour
+
+The `$hour` date operator is used to extract the exact hour the document was created on. The output goes from 0-23.
+
+Example:
+```js
+db.pokedex.aggregate([
+  { $match: { name: "Bulbasaur" } },
+  {
+    $project: {
+      Hour: { $hour: "$Date" },
+      _id: 0
+    }
+  }
+]);
+
+```
+
+Output:
+```js
+{ 
+  "Hour" : 7 
```

Comment:
Also in the results, why capitalized fields?

Reply:
```suggestion
  "createdAtHour" : 7 
```

```suggestion
  "hourOfCreation" : 7 
```

Reply:
Should probably do this for all results

Reply:
Which one? hourOfCreation / createdAtHour

Reply:
both sound good. choose the one that you like but make sure to keep consistent throughout the workout (find if there are any other instances of `'Hour'` and change them to your selection).

Reply:
I put capitalized because the names were similar to the operators. Just so people can differentiate them more easily. Should I change the Minutes to `createdAtMinute` and any other similar operator?

Reply:
I'll change it. It makes more sense that way.
createdAt= full ISODate
createdAt(Operator) as if you're taking a part from `createdAt`